manager/database.py

```diff
@@ -28,26 +32,42 @@ def add_url(self, url):
         """
         url will be adding to database.
         """
+
         try:
             self.url = url
-            date = datetime.date.today()
-            start = datetime.datetime.now()
-            time = start.strftime("%H:%M:%S")
+            global date
+            global time
             self.cursor.execute('''
-            INSERT INTO bookmarks(url, date, time) VALUES (?, ?, ?)
-            ''', (self.url, date, time))
+            INSERT INTO bookmarks(url, tags, date, time) VALUES (?, ?, ?, ?)
+            ''', (self.url, None, date, time))
             self.db.commit()
         except Exception as e1:
             print("URL is already present in database.", e1)
 
+    def tag_url(self, tag_name, tagged_url):
+        """
+        URLs can be added by respective Tags.
+        """
+        self.tag = tag_name
+        self.url = tagged_url
+        try:
+            global date
+            global time
+            self.cursor.execute(
+                '''INSERT INTO bookmarks(url, tags, date, time)
+                VALUES(?, ?, ?, ?)''', (self.url, self.tag, date, time))
```

Comment:
@ganeshhubale I don't think it will take current time value. Once the file is called that time it will take the time value.

Then for insertion, it will take the same value. I'll suggest move 
```
time = start.strftime("%H:%M:%S")
```
to this method. So it will get the exact time at the insertion.

Reply:
@amolkahat Please review.
- It takes current time while insertion. 
- I have already declared `time = start.strftime("%H:%M:%S")` as global variable. So that I can use it in two functions  **add_url** and **tag_url**.

Reply:
@ganeshhubale I don't think you understand my solution correctly.

What I'm trying to say if you are trying to access it globally then it will take a constant value. So I suggested you to move it to the method.

for ex:
```import datetime
import time as t

start = datetime.datetime.now()
time = start.strftime("%H:%M:%S")



class hello_ex(object):
    def __init__(self):
        t.sleep(10)

    def display_time(self):
        print(time)


if __name__ == '__main__':
    print(time)
    t.sleep(10)
    print(time)
    h = hello_ex()
    h.display_time()
```
The output of this script is like:
```
17:12:16
17:12:16
17:12:16
```
It will display same time after each 10s.

Reply:
Yes. I understood. 
I have made changes as per suggestions to `database.py` file. Please review.
- modified script:
```
import time as t
import datetime as dt

class hello_ex(object):
    def __init__(self):
        t.sleep(10)

    def display_time(self):
        start = dt.datetime.now()
        time = start.strftime("%H:%M:%S")

        print(time)


if __name__ == '__main__':
    start = dt.datetime.now()
    time = start.strftime("%H:%M:%S")

    print(time)
    t.sleep(10)

    start = dt.datetime.now()
    time = start.strftime("%H:%M:%S")

    print(time)
    h = hello_ex()
    h.display_time()

```

Output of above script:
```
18:47:32
18:47:42
18:47:52
```